code/obj/item/clothing/suits.dm

```diff
@@ -1283,32 +1283,55 @@
 	item_state = "space-neon"
 
 /obj/item/clothing/suit/space/custom // Used for nanofabs
+	icon_state = "spacemat"
+	inhand_image_icon = "s_suit"
+	item_state = "spacemat"
 	name = "bespoke space suit"
-	desc = "A suit that protects against low pressure environments, custom-made just for you!"
-	onMaterialChanged()
-		if(src.material)
-			if(material.hasProperty("thermal"))
-				var/prot = 100 - material.getProperty("thermal")
+	desc = "A custom built suit that protects your soft tissues from hard vacuum."
+	proc/setupSuitProp(var/datum/material/F, var/datum/material/R) // f is fabric, decides protection, R is reinforcment, decides melee prot
+
+		if(F)
+			if(F.hasProperty("thermal"))
+				var/prot = 100 - F.getProperty("thermal")
 				setProperty("coldprot", prot)
 				setProperty("heatprot", round(prot/2))
 			else
 				setProperty("coldprot", 30)
 				setProperty("heatprot", 15)
 
-			if(material.hasProperty("permeable"))
-				var/prot = 100 - material.getProperty("permeable")
+			if(F.hasProperty("permeable"))
+				var/prot = 100 - F.getProperty("permeable")
 				setProperty("viralprot", prot)
 			else
 				setProperty("viralprot", 40)
 
-			if(material.hasProperty("density"))
-				var/prot = round(material.getProperty("density") / 13)
-				setProperty("meleeprot", prot)
+			if(F.hasProperty("density")) // for RANGED
+				var/prot = round(F.getProperty("density") / 13)
 				setProperty("rangedprot", (0.2 + round(prot/10, 0.1)))
 			else
-				setProperty("meleeprot", 2)
 				setProperty("rangedprot", 0.4)
 
+			if(R.hasProperty("hard")) // for MELEE
+				var/prot = round(R.getProperty("hard") / 13)
+				if(prot < 3)// most metals are very soft, so for balance's sake it'll be (usually) at least that of normal suits
+					setProperty("meleeprot", 3)
+				if(prot > 6 )
+					setProperty("meleeprot", 6) // that having been said, we don't want anything more powerful than security's armor
+				setProperty("meleeprot", prot)
+			else
+				setProperty("meleeprot", 2)
+
+			if(R.hasProperty("density"))
+				var/clunk = R.getProperty("density")
+				if (clunk <= 15) // lighter metals = faster
+					setProperty("space_movespeed", 0.4) // since movespeed is already initalized, no need to have an else
+				else if (clunk >= 40)
+					setProperty("space_movespeed", 0.7) // .1 above normal spacesuits
+	proc/setsuitname(var/datum/material/F, var/datum/material/R)
+		if (F && R)
+			name = "[R]-reinforced [F] bespoke space suit"
+		else if (F)
+			name = " [F] bespoke space suit"
```

Comment:
This would perhaps be better written as the override of the `UpdateName` proc which would then use `src.material` and `src.someNewVarContainingTheOtherMaterial`. Then you'd call `UpdateName` from the second-material setting proc. Assuming you call parent in `onMaterialChanged` that should handle calling `UpdateName` from there. If you don't call the parent, well, then this all might run into some weird issues, I'm unsure off the top of my head. However, if you decide not calling parent is what you need you will need to call it from `onMaterialChanged` explicitly too.